### PR TITLE
Updated links on Firefox workflow overview

### LINKED ISSUE #350: Update links on Firefox workflow overview
- [x] Code your extension (currently results in 404) 
- [x] Promote your extension
- [x] Featured extensions
- [x] Update your extension
- [x] Sideloading add-ons (currently results in 404)
- [x] Desktop apps (currently results in 404)
- [x] Enterprise (currently results in 404)

### PR DESCRIPTION
**Pull request by ruchikabgosain**
This fixes #350 in the file `firefox-workflow-overview.md`

Fixes the 404 links to : 
- Sideloading add-ons 
- Desktop apps 
- Enterprise 

Added the links to: 
-  Promote your extension
-  Update your extension

CC: @caitmuenster
Also, I couldn't find an appropriate link for `Featured extensions`, let me know how to fix it!

### CODE REVIEW

**Comment by caitmuenster** [comment]
@ruchikabgosain No worries -- we're replacing Featured Extensions with Recommended Extensions and haven't written the developer facing doc yet. For the moment, can you replace the word "featured" with "recommended" and link to https://blog.mozilla.org/addons/2019/04/08/recommended-extensions-program-coming-soon/?

**Comment by ruchikabgosain** [comment]
Incorporated these, @caitmuenster let me know if this works now?

**Comment by ruchikabgosain** [comment]
Hope it's fine now @caitmuenster , there was some messing up with the `href` tag!

**Comment by caitmuenster** [comment]
@ruchikabgosain I'm wrapping up for the day but this is on my list for tomorrow.

**Comment by ruchikabgosain** [comment]
Totally @caitmuenster hope you had a good day 🌙

File: documentation/develop/firefox-workflow-overview.md [hunk 1]
```diff
@@ -52,33 +52,33 @@ date: 2019-05-20 03:25:40
         <td>Choose your IDE or code editor</td>
         <td>Run your extension with <a href="/documentation/develop/getting-started-with-web-ext#testing-out-an-extension">web-ext run</a> or <a href="/documentation/develop/temporary-installation-in-firefox/">about:debugging</a></td>
         <td>Create an <a href="https://addons.mozilla.org">addons.mozilla.org</a>account</td>
-        <td>Promote your extension</td>
+      <td><a href="/documentation/publish/promoting-your-extension/"Promote your extension</a></td>
```
Comment (caitmuenster) [inline documentation/develop/firefox-workflow-overview.md]: Ah, it looks like the problem here is that you need a `>` at the end of <a href="/documentation/publish/promoting-your-extension/".

Comment (ruchikabgosain) [reply documentation/develop/firefox-workflow-overview.md]: Fixed this! 😸

**Pull request merged**